programs/clearing_house/src/state/bank.rs

```diff
@@ -33,23 +37,68 @@ pub struct Bank {
     pub maintenance_asset_weight: u128,
     pub initial_liability_weight: u128,
     pub maintenance_liability_weight: u128,
+    pub imf_factor: u128,
 }
 
 impl Bank {
-    pub fn get_asset_weight(&self, margin_requirement_type: &MarginRequirementType) -> u128 {
-        match margin_requirement_type {
-            MarginRequirementType::Initial => self.initial_asset_weight,
-            MarginRequirementType::Partial => self.maintenance_asset_weight,
+    pub fn get_asset_weight(
+        &self,
+        size: u128,
+        margin_requirement_type: &MarginRequirementType,
+    ) -> ClearingHouseResult<u128> {
+        let size_precision = 10_u128.pow(self.decimals as u32);
+
+        let size_in_amm_reserve_precision = if size_precision > AMM_RESERVE_PRECISION {
+            size / (size_precision / AMM_RESERVE_PRECISION)
+        } else {
+            (size * AMM_RESERVE_PRECISION) / size_precision
```

Comment:
nit: `size * (AMM_RESERVE_PRECISION / size_precision)` is probably more efficient with compute units

Reply:
does that break if the bank decimals are >13?

Reply:
ah true